### PR TITLE
fix: fake clientset not using the correct group name

### LINKED ISSUE #3517: The fake clientset does not work as expected
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Trying to use the fake client set to write some unit tests:
```
kongClientset := kongtestclient.NewSimpleClientset(&kongv1.KongPlugin{
	ObjectMeta: metav1.ObjectMeta{
		Name:      "test-plugin",
		Namespace: "test-ns",
	},
})
```

Where `kongv1` is the following import: `kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"`

Calling 
```
plugin, err := kongClientset.ConfigurationV1().KongPlugins("test-ns").Get(ctx, "test-plugin", metav1.GetOptions{})
```
returns an error that the plugin can not be found. This is because the group of the fake generated resources is wrong: https://github.com/Kong/kubernetes-ingress-controller/blob/main/pkg/clientset/typed/configuration/v1/fake/fake_kongplugin.go#L39. It should be `configuration.konghq.com` instead of `configuration` 

It looks like `client-gen` uses the folder structure to determine the group and at the moment it's set to `configuration`: https://github.com/Kong/kubernetes-ingress-controller/tree/main/pkg/apis/configuration

Renaming the folder to `configuration.konghq.com` and reruning `client-gen` generates the correct groups. I am not sure if having the wrong name for the folder has other impacts I've only noticed this when trying to use it for testing. 

Renaming that folder seems to break the public API though so I was not sure if it was an acceptable solution. I haven't seen any ways around it with the `client-gen` options

### Expected Behavior

The fake simple client set can be used for testing.

### Steps To Reproduce

```markdown
1. Try to use the `NewSimpleClientSet` in tests as described in the current behaviour
```

### Kong Ingress Controller version

```shell
v2.8.1
```

### Kubernetes version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by robertmarsal**
**What this PR does / why we need it**:

This PR fixes the group mismatch reported in https://github.com/Kong/kubernetes-ingress-controller/issues/3517

**Which issue this PR fixes**:

fixes #3517 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by pmalek** [comment]
@robertmarsal This looks pretty good. I wonder why the placement of kubebuilder tags is relevant and where is that mentioned in the docs 🤔 

In any case I'd consider this as reviewable, just some small suggestions:

- Please add a changelog entry since that is relevant enough that some users might be interested in this change

- A small UT (or more than 1) could be beneficial to ensure we don't break this functionality in the future. I don't have a strong opinion on this but I think that putting this in `pkg/apis/configuration/...` makes sense.

  Something like:
  
  ```go
  package telemetry
  
  import (
	  "context"
	  "testing"
  
	  "github.com/stretchr/testify/require"
	  metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
  
	  configurationv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
	  kongfake "github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset/fake"
  )
  
  func TestFakePlugin(t *testing.T) {
	  cl := kongfake.NewSimpleClientset(&configurationv1.KongPlugin{
		  ObjectMeta: metav1.ObjectMeta{
			  Name:      "test-plugin",
			  Namespace: "test-ns",
		  },
	  })
  
	  plugin, err := cl.ConfigurationV1().KongPlugins("test-ns").
		  Get(context.Background(), "test-plugin", metav1.GetOptions{})
	  require.NoError(t, err)
	  require.NotNil(t, plugin)
	  require.Equal(t, "test-plugin", plugin.Name)
  }
  ```
  
  should do.

**Comment by pmalek** [comment]
And please sign the CLA so that we can merge it and you can earn the contribution :)

**Comment by robertmarsal** [comment]
Thanks @pmalek I've updated the PR with your suggestions. I had to put the test into a separate package due to import cycles. I didn't want to put the tests in the clientset package since that's generated code.

File: pkg/clientset_test/clientset_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package clientset_test
+
+import (
+	"context"
+	"testing"
+
+	configurationv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+	kongfake "github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset/fake"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (pmalek) [inline pkg/clientset_test/clientset_test.go:10]: Nit import order:

```suggestion
	"github.com/stretchr/testify/require"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	
	configurationv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
	kongfake "github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset/fake"
```

**Review by pmalek — COMMENTED** [review]
Just one nit.

Can you please rebase? We've added a change which should skip the enterprise tests (which require license secret which is not available for PRs from forks). This should make the check green.

**Comment by robertmarsal** [comment]
Of course. All done now. Thanks

File: pkg/clientset_test/clientset_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package clientset_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
```
Comment (pmalek) [inline pkg/clientset_test/clientset_test.go:9]: Not sure why but linter is complaining about this line 🤔 

https://github.com/Kong/kubernetes-ingress-controller/actions/runs/4297670409/jobs/7494491099#step:7:40

Do you mind checking this locally via `make lint` and `make verify.generators` to ensure we're good? Sorry for this to be a pain.

Comment (robertmarsal) [reply pkg/clientset_test/clientset_test.go]: There was some whitespace there. I've removed it and waiting for the build status now

**Review by pmalek — APPROVED** [review]
This looks good. Thanks once again! Feel free to submit this contribution through [this form](https://docs.google.com/forms/d/e/1FAIpQLSfacQHac-PYIDIrDOi_W2l7cWxBDRIVBs6xE5HnlojPvdhL9g/viewform) to get KIC contributor t-shirt

**Pull request merged**